src/foam/nanos/dig/SUGAR.js

```diff
@@ -0,0 +1,256 @@
+/**
+ * @license
+ * Copyright 2018 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.dig',
+  name: 'SUGAR',
+
+  documentation: 'SUGAR : Service Unified GAteway Relay - Perform non-DAO operations against a web service',
+
+  tableColumns: [
+    'id',
+    'serviceKey',
+    'method'
+  ],
+
+  requires: [
+    'foam.nanos.dig.Argument'
+  ],
+
+  css: `
+    .property-argumentInfo button {
+      display: none;
+    }
+  `,
+
+  properties: [
+    {
+      class: 'String',
+      name: 'id',
+      displayWidth: 40
+    },
+    {
+      class: 'String',
+      name: 'serviceKey',
+      label: 'Service',
+      documentation: 'non DAOs list as service',
+      view: function(_, X) {
+        var E = foam.mlang.Expressions.create();
+        return foam.u2.view.ChoiceView.create({
+          dao: X.nSpecDAO
+            .where(E.AND(E.EQ(E.ENDS_WITH(foam.nanos.boot.NSpec.ID, 'DAO'), false), E.EQ(foam.nanos.boot.NSpec.SERVE, true)))
+            .orderBy(foam.nanos.boot.NSpec.ID),
+          objToChoice: function(nspec) {
+            return [nspec.id, nspec.id];
+          }
+        });
+      },
+      postSet: function() {
+        var service = this.__context__[this.serviceKey];
+
+        if ( ! service ) return;
+
+        if ( ! service.cls_.getAxiomByName('delegate') ) return;
+
+        var of = this.lookup(service.cls_.getAxiomByName('delegate').of);
+
+        if ( ! of ) return;
+
+        this.interfaceName = of.id.toString();
+
+        var methods = of.getOwnAxiomsByClass(foam.core.Method);
+        //var methodName = methods.map(function(m) { return m.name; }).sort();
+
+        var filteredMethod =
+          methods.filter(function(fm) {
+            for ( var j = 0 ; j < fm.args.length ; j++ ) {
+               if ( fm.args[j].javaType.toString() == "foam.core.X" ) {
+                  return false;
+               }
+               return true;
+             }
+          }).map(function(m) { return m.name; }).sort();
+
+        if ( filteredMethod.length > 0 ) {
+          for ( var i = 0 ; i < methods.length ; i++ ) {
+            if ( methods[i].name == filteredMethod[0] ) {
```

Comment:
Not sure if we support overriding on services' method, if we do, filtering would be a problem just by name. 
If we don't, why don't you just break the loop once you find the correct method with the same name as filteredmethod?

Reply:
`array.find` would work here if you are trying to search a **single** method by name.